File: array/1909_remove_one_element_to_make_the_array_strictly_increasing/1909_remove_one_element_to_make_the_array_strictly_increasing.go
package main

// 1,2,3,5,4,6,
// 1,2,3,5,2,6

func canBeIncreasing(nums []int) bool {
	chance := true
	for i := 0; i < len(nums)-1; i++ {
		if nums[i] >= nums[i+1] {
			if chance == false {
				// no chance, return false
				return false
			}
			if i > 0 {
				// we have to remove nums[i+1] if nums[i+1] ≤ nums[i-1]
				if nums[i+1] <= nums[i-1] {
					nums[i+1] = nums[i]
				}
			}
			chance = false
		}
	}
	return true
}

File: array/1909_remove_one_element_to_make_the_array_strictly_increasing/1909_remove_one_element_to_make_the_array_strictly_increasing.py
class Solution:
    def canBeIncreasing(self, nums: List[int]) -> bool:
        for i in range(len(nums)):
            tmp_nums = nums[0:i] + nums[i+1:]
            for j in range(len(tmp_nums)-1):
                if tmp_nums[j] >= tmp_nums[j+1]:
                    break
            else:
                return True
        return False



nums = [1,2,10,5,7]
res = Solution().canBeIncreasing(nums)
print(res)